backend/api/Controllers/MissionSchedulingController.cs

```diff
@@ -313,54 +312,6 @@ [FromBody] ScheduledMissionQuery scheduledMissionQuery
 
             var missionTasks = await missionLoader.GetTasksForMission(missionSourceId);
 
-            List<Area?> missionAreas;
-            missionAreas = missionTasks
-                .Where(t => t.TagId != null)
-                .Select(t =>
-                    stidService.GetTagArea(t.TagId!, scheduledMissionQuery.InstallationCode).Result
-                )
-                .ToList();
-
-            var missionInspectionAreaNames = missionAreas
-                .Where(a => a != null)
-                .Select(a => a!.InspectionArea.Name)
-                .Distinct()
-                .ToList();
-            if (missionInspectionAreaNames.Count > 1)
-            {
-                string joinedMissionInspectionAreaNames = string.Join(
-                    ", ",
-                    [.. missionInspectionAreaNames]
-                );
-                logger.LogWarning(
-                    "Mission {missionDefinition} has tags on more than one inspection area. The inspection areas are: {joinedMissionInspectionAreaNames}.",
-                    missionDefinition.Name,
-                    joinedMissionInspectionAreaNames
-                );
-            }
-
-            var sortedAreas = missionAreas
-                .GroupBy(i => i)
-                .OrderByDescending(grp => grp.Count())
-                .Select(grp => grp.Key);
-            var area = sortedAreas.First();
-
-            if (area == null && sortedAreas.Count() > 1)
-            {
-                logger.LogWarning(
-                    "Most common area in mission {missionDefinition} is null. Will use second most common area.",
-                    missionDefinition.Name
-                );
-                area = sortedAreas.Skip(1).First();
-            }
-            if (area == null)
-            {
-                logger.LogError(
-                    $"Mission {missionDefinition.Name} doesn't have any tags with valid area."
-                );
-                return NotFound($"No area found for mission '{missionDefinition.Name}'.");
-            }
```

Comment:
Could we perhaps keep this block of code in order to find which inspection area to assign the mission definition? We do not need to have an Area model in the database to do this, as it fetches them from an external API (STID). We do however need the inspection area to have an understanding of STID areas though.

Reply:
Originally we planned InspectionAreas to be a list or graph of Areas. Could we potentially store the references to STID areas as a list in InspectionAreas still, but without storing them as Area objects, for instance by only storing the area name?